### PR TITLE
Add RPM image scanner

### PR DESCRIPTION
**Pull request by micahhausler**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Adds support for scanning RPM databases

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

I split layer scanning tests into separate files

#### Does this PR introduce a user-facing change?

```release-note
Added support for scanning images with RPM package managers
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
That's pretty cool, thank you for the contribution!

@puerco PTAL
/hold

**Comment by micahhausler** [comment]
I rebased to include the fix in #343

File: pkg/osinfo/scanner_rpm.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		}
+
+		packages = append(packages, PackageDBEntry{
+			Package:      p.Name,
+			Version:      p.Version,
+			Architecture: p.Arch,
+			Type:         "rpm",
+			// Namespace is set later
+			MaintainerName: p.Vendor,
+			// Most RPM pacakges don't have SPDX-valid license names
```
Comment (puerco) [inline pkg/osinfo/scanner_rpm.go:127]: Curious: Is this true for all distros? Are they using an alternative naming scheme that could be translated later?

File: pkg/osinfo/container_scanner.go [hunk 2]
```diff
@@ -56,6 +56,8 @@ func ReadOSPackages(layers []string) (
 		cs = newDebianScanner()
 	case OSAlpine, OSWolfi:
 		cs = newAlpineScanner()
+	case OSAmazonLinux, OSFedora, OSRHEL:
```
Comment (puerco) [inline pkg/osinfo/container_scanner.go:59]: nit for later: we can also add opensuse ! 🦎
(Not actual chameleon emoji, but close :)

**Review by puerco — APPROVED** [review]
This is awesome @micahhausler !

The oldest branch I had on my machine was work in progress to implement this. It has finally happened, thank you Micah :smiling_face_with_tear: 

I left a question there, but in it goes! I think it is time to cut a new relase :rocket:

File: pkg/osinfo/scanner_rpm.go (hunk 1 above)
Comment (micahhausler) [reply pkg/osinfo/scanner_rpm.go:127]: This potentially affects other distros, I haven't looked at APK or DEB for validation. I spent a _lot_ of time digging into this and found that basically the story is not great:
* If you stuff a SPDX-invalid license name into the field, the SPDX doc fails to verify
* Some distros like [Fedora](https://fedoraproject.org/wiki/Changes/SPDX_Licenses_Phase_2) are working to migrate license names to SPDX license strings. This effort is not complete yet for Fedora.  Fedora has a [`license-fedora2spdx`](https://pagure.io/copr/license-validate) package (Python scripts/ config) that tries to do convert license name strings, but fails pretty hard in a lot of cases. Ex: `BSD` was Fedora-valid, but the tool spits out something like `"BSD": "Warning: more options on how to interpret BSD. Possible options: ['BSD-1-Clause', 'BSD-2-Clause-FreeBSD', 'BSD-2-Clause-Views', 'BSD-2-Clause', 'BSD-3-Clause-Modification', 'BSD-3-Clause']` which requires more info. 
* There isn't a documented way to parse the RPM database to produce the equivalent of `rpm -q --licensefiles $PACKAGE` and get the list of license files in the image. 
  * Most distros seem to put license files in `/usr/share/licenses/$PACKAGE` with multiple different filenames (Try running `find /usr/share/licenses -type f -exec basename {} \; | sort |uniq | wc -l` in any RPM-based container distro, you'll get >40 different names). Even if you ran `rpm -qL $PKG`, or just searched in the directory and tried to analyze the license files, you wouldn't know if there's an `AND` vs an `OR` for a given package. (`AND` typically indicating valid SPDX license vs `and` being a legacy license name. Try `rpm -qa --queryformat '%{name} %{license}\n' | grep -E ' (and|or) '` to find those legacy ones. )

We _could_ try calling [`Reader.LicenseFromLabel()`](https://github.com/micahhausler/bom/blob/rpm-scanner/pkg/license/license.go#L153:L153) and only including valid ones, but for now I just mirrored what DEB and APK parsers do.

**Pull request merged**